src/ingress_definition.py

```diff
@@ -316,16 +316,54 @@ def proxy_connect_timeout(self) -> str:
 
     @property
     def rewrite_enabled(self) -> bool:
-        """Return whether rewriting should be enabled from config or relation."""
-        value = self._get_config_or_relation_data("rewrite-enabled", True)
+        """
+        Return whether rewriting should be enabled from config, per-app ingress
+        relation (strip_prefix), or nginx-route relation, in that order.
+        """
+        value = self._get_config("rewrite-enabled")
         # config data is typed, relation data is a string
         # Convert to string, then compare to a known value.
-        return str(value).lower() == "true"
+        if value is not None:
+            return str(value).lower() == "true"
+
+        if self.is_ingress_relation:
+            try:
+                return (
+                    cast(IngressPerAppProvider, self.ingress_provider)
+                    .get_data(self.relation)
+                    .app.strip_prefix
+                )
+            except DataValidationError as exc:
+                raise InvalidIngressError(msg=f"{exc}, cause: {exc.__cause__!r}") from exc
+
+        value = self._get_relation("rewrite-enabled")
+        if value is not None:
+            return str(value).lower() == "true"
+
+        return False
 
     @property
     def rewrite_target(self) -> str:
-        """Return the rewrite target from config or relation."""
-        return cast(str, self._get_config_or_relation_data("rewrite-target", "/"))
+        """Return the rewrite target from config, per-app ingress
+        relation (strip_prefix), or nginx-route relation, in that order."""
+        value = self._get_config("rewrite-target")
+        if value is not None:
```

Comment:
```suggestion
        if value:
```
I think we can't use `is` here because the default value of the `rewrite-target` configuration is `""`.